### ISSUE TITLE
add method to check macro is defined

### ISSUE THREAD

**Issue by changloong** [twigphp/Twig#1313]
code:

``` php
{% macro display(object) %}
    {% import "AppWebBundle:Invest:admin.macro.html.twig" as parent_macro %}
    {% if parent_macro.display is defined %}
        {{ parent_macro.display(object)}}
    {% else %}
        {{ object }}
    {% endif %}
{% endmacro %} 
```

throw error:

``` sh
The "defined" test only works with simple variables in invest.admin.html.twig at line 5
```

**Comment by SpacePossum** [comment]
https://github.com/twigphp/Twig/pull/490
(link update, other is in old repo which now serves 404)

**Comment by fabpot** [comment]
implemented in #3014
